app/controllers/concerns/foreman/controller/auto_complete_search.rb

```diff
@@ -9,7 +9,7 @@ module Foreman::Controller::AutoCompleteSearch
   def auto_complete_search
     begin
       model = (controller_name == "hosts") ? Host::Managed : model_of_controller
-      @items = model.complete_for(params[:search], {:controller => controller_name})
+      @items = model.complete_for(params[:search], {:controller => controller_name, :enhanced_filter => {:has_column => "organization_id", :filter => {:organization_id => Organization.current.id}}})
```

Comment:
Katello has its own auto_complete_search definition, this change would need to happen there as well https://github.com/Katello/katello/blob/16440c3907d6b1f64c6edc2a2dedc213085436e0/app/controllers/katello/concerns/filtered_auto_complete_search.rb#L8 . Things like Content > Subscriptions and search for "name = " still offers subscriptions from other organizations.

Reply:
@Thorben-D can you please have a look at the last comments?